services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -25,17 +25,20 @@ import (
 	"os"
 	"path/filepath"
 	"syscall"
+	"time"
 )
 
 type metrics struct {
-	sizeOnDisk *metric.Gauge
+	sizeOnDisk          *metric.Gauge
+	indexLastUpdateTime *metric.Gauge
 }
 
 const blocksFilename = "blocks"
 
-func newMetrics(m metric.Factory) *metrics {
+func generateBlockStorageMetrics(m metric.Factory) *metrics {
 	return &metrics{
-		sizeOnDisk: m.NewGauge("BlockStorage.FileSystemSize.Bytes"),
+		sizeOnDisk:          m.NewGauge("BlockStorage.FileSystemSize.Bytes"),
+		indexLastUpdateTime: m.NewGauge("BlockStorage.FileSystemIndex.LastUpdateTime"),
```

Comment:
should be instantiated with the current timestamp value not zero

Reply:
why? building the index occurs immediately after init (init -> open blocks file -> build index), and on build index or any append block current timestamp is used

Reply:
a possible race condition. we should at least make sure that we don't ever Error when the value is 0